### PR TITLE
add view for Employment Details EOY

### PR DESCRIPTION
**Pull request by michael-rob-poole**
### Description
Add EOY employment view - Change links and Save and continue button added

Add a link to the relevant story in Jira
[SASS-1010](https://jira.tools.tax.service.gov.uk/browse/SASS-768)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of master?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current master that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of master?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: app/controllers/employment/CheckEmploymentDetailsController.scala [hunk 1]
```diff
@@ -18,43 +18,62 @@ package controllers.employment
...
-                                                 implicit val ec: ExecutionContext) extends FrontendController(cc) with I18nSupport with SessionHelper {
+                                                 ec: ExecutionContext) extends FrontendController(cc) with I18nSupport with SessionHelper {
 
 
   def show(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
 
     def result(allEmploymentData: AllEmploymentData): Result = {
 
-      val source: Option[EmploymentSource] = allEmploymentData.hmrcEmploymentData.find(source => source.employmentId.equals(employmentId))
+      val customerData: (Option[EmploymentSource], Boolean) = getCustomerData(allEmploymentData, employmentId, taxYear)
```
Comment (BenCosfordHMRC) [inline app/controllers/employment/CheckEmploymentDetailsController.scala:48]: I'd change customerData to be only `allEmploymentData.customerEmploymentData.find(source => source.employmentId.equals(employmentId))`

and then when you do the if statements on line 50 use `if(customerData.isDefined && !inYearAction.inYear(taxYear))` instead.

could define a val like isUsingCustomerData that equals `customerData.isDefined && !inYearAction.inYear(taxYear)`

File: conf/application.conf [hunk 2]
```diff
@@ -171,7 +171,7 @@ defaultTaxYear = 2022
 timeoutDialogTimeout = 900
 timeoutDialogCountdown = 120
 
-taxYearErrorFeatureSwitch = true
+taxYearErrorFeatureSwitch = false
```
Comment (BenCosfordHMRC) [inline conf/application.conf:174]: swap back to true

File: app/views/employment/CheckEmploymentDetailsView.scala.html [hunk 3]
```diff
@@ -31,54 +28,91 @@
...
-@(employerNameData: String, employerRefData: Option[String], employmentDetails: Option[EmploymentData], taxYear: Int)(implicit user: User[AnyContent], messages: Messages, appConfig: AppConfig)
+@(employerNameData: String, employerRefData: Option[String], employmentDetails: Option[EmploymentData], taxYear: Int, isInYear:Boolean, employmentId:String, isCustomerData:Boolean)(implicit user: User[AnyContent], messages: Messages, appConfig: AppConfig)
 
 
 @headingForUse = @{
     messages(s"employment.employmentDetails.heading.${if(user.isAgent) "agent" else "individual"}")
 }
 
+@margin = @{
+    if(!isCustomerData) "govuk-!-margin-bottom-2" else "govuk-!-margin-bottom-4"
```
Comment (BenCosfordHMRC) [inline app/views/employment/CheckEmploymentDetailsView.scala.html:39]: are the difference in margins needed to match the prototype?

File: app/views/employment/CheckEmploymentDetailsView.scala.html [hunk 4]
```diff
@@ -31,54 +28,91 @@
...
+                        actions = if(!isInYear) Seq((changeLink, messages("common.change"), Some(hiddenText))) else Seq.empty
+                    )
+                }
             )
         )
-
+    }
+    @if(!isInYear){
+        @formWithCSRF(action = controllers.employment.routes.CheckEmploymentDetailsController.submit(taxYear, employmentId)) {
+            @button("common.saveAndContinue", classes = Some("govuk-!-margin-top-6"))
```
Comment (BenCosfordHMRC) [inline app/views/employment/CheckEmploymentDetailsView.scala.html:114]: again is this margin needed to match the prototype?

**Review by BenCosfordHMRC — CHANGES_REQUESTED** [review]

File: app/views/employment/CheckEmploymentDetailsView.scala.html (hunk 4 above)
Comment (michael-rob-poole) [reply app/views/employment/CheckEmploymentDetailsView.scala.html:114]: Yep. I copied this from InterestCYA

File: app/views/employment/CheckEmploymentDetailsView.scala.html (hunk 3 above)
Comment (michael-rob-poole) [reply app/views/employment/CheckEmploymentDetailsView.scala.html:39]: I couldn't find an example in the prototype where there wasn't paragraph text. For when there wasn't <p> text margin-2 did look kind of ridiculous though

File: app/controllers/employment/CheckEmploymentDetailsController.scala (hunk 1 above)
Comment (michael-rob-poole) [reply app/controllers/employment/CheckEmploymentDetailsController.scala:48]: agreed -changed

**Review by BenCosfordHMRC — APPROVED** [review]

**Comment by BenCosfordHMRC** [comment]
Screenshot 2021-07-06 at 13 37 57

Few dependencies can be bumped up

**Pull request merged**